middlewares/prometheus.go

```diff
@@ -62,28 +63,56 @@ func NewPrometheus(name string, config *types.Prometheus) *Prometheus {
 
 	hv := stdprometheus.NewHistogramVec(
 		stdprometheus.HistogramOpts{
-			Name:        latencyName,
+			Name:        reqDurationName,
 			Help:        "How long it took to process the request.",
 			ConstLabels: stdprometheus.Labels{"service": name},
 			Buckets:     buckets,
 		},
 		[]string{},
 	)
+	hv = registerHistogramVec(hv)
+	prom.reqDurationHistogram = prometheus.NewHistogram(hv)
+	collectors = append(collectors, hv)
+
+	cv = stdprometheus.NewCounterVec(
+		stdprometheus.CounterOpts{
+			Name:        retriesTotalName,
+			Help:        "How many request retries happened in total.",
+			ConstLabels: stdprometheus.Labels{"service": name},
+		},
+		[]string{},
+	)
+	cv = registerCounterVec(cv)
+	prom.retryCounter = prometheus.NewCounter(cv)
+	collectors = append(collectors, cv)
+
+	return &prom, collectors
+}
+
+func registerCounterVec(cv *stdprometheus.CounterVec) *stdprometheus.CounterVec {
+	err := stdprometheus.Register(cv)
 
-	err = stdprometheus.Register(hv)
 	if err != nil {
 		e, ok := err.(stdprometheus.AlreadyRegisteredError)
 		if !ok {
 			panic(err)
 		}
-		m.latencyHistogram = prometheus.NewHistogram(e.ExistingCollector.(*stdprometheus.HistogramVec))
-	} else {
-		m.latencyHistogram = prometheus.NewHistogram(hv)
+		cv = e.ExistingCollector.(*stdprometheus.CounterVec)
 	}
 
-	return &m
+	return cv
 }
 
-func (p *Prometheus) handler() http.Handler {
-	return promhttp.Handler()
+func registerHistogramVec(hv *stdprometheus.HistogramVec) *stdprometheus.HistogramVec {
+	err := stdprometheus.Register(hv)
+
+	if err != nil {
+		e, ok := err.(stdprometheus.AlreadyRegisteredError)
+		if !ok {
+			panic(err)
```

Comment:
Ok, I rebuilt it now in a way that if any of the Prometheus Metrics can not be registered, an error will be returned and the `*Prometheus` will be `nil`. This will lead to no Metrics instrumentation and an log message on Error level. Does it make sense to you this way?